src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -0,0 +1,193 @@
+
+        [Test]
+        public void Should_make_queues_non_transactional_if_requested()
+        {
+            const bool isTransactional = false;
+
+            var testQueueName = "MsmqQueueCreatorTests.txreceiver";
+
+            DeleteQueueIfPresent(testQueueName);
+
+            var creator = new QueueCreator(isTransactional);
+            var bindings = new QueueBindings();
+
+            bindings.BindReceiving(testQueueName);
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            var queue = GetQueue(testQueueName);
+
+            Assert.AreEqual(isTransactional, queue.Transactional);
+        }
+
+        [Test]
+        public void Msmq_will_add_everyone_and_anonymous_by_default()
+        {
+            var testQueueName = "MsmqQueueCreatorTests.MsmqDefaultPermissions";
+
+            DeleteQueueIfPresent(testQueueName);
+
+            var path = MsmqAddress.Parse(testQueueName).PathWithoutPrefix;
+
+            using (var queue = MessageQueue.Create(path))
+            {
+                MessageQueueAccessRights? everyoneAccessRights;
+
+                Assert.True(queue.TryGetPermissions(LocalEveryoneGroupName, out everyoneAccessRights));
+                Assert.True(everyoneAccessRights.HasValue, "User should have access rights");
+                Assert.AreEqual(MessageQueueAccessRights.GenericWrite, everyoneAccessRights, "Msmq should give 'everyone' write access by default");
+
+
+                MessageQueueAccessRights? anonymousAccessRights;
+
+                Assert.True(queue.TryGetPermissions(LocalAnonymousLogonName, out anonymousAccessRights));
+                Assert.True(anonymousAccessRights.HasValue, "User should have access rights");
+                Assert.AreEqual(MessageQueueAccessRights.WriteMessage, anonymousAccessRights, "Msmq should give 'anonymous' write access by default");
+            }
+        }
+
+
+        [Test]
+        public void Should_not_add_everyone_and_anonymous_to_already_existing_queues()
+        {
+            var testQueueName = "MsmqQueueCreatorTests.NoChangesToExisting";
+
+            DeleteQueueIfPresent(testQueueName);
+
+            var path = MsmqAddress.Parse(testQueueName).PathWithoutPrefix;
+
+            using (var existingQueue = MessageQueue.Create(path))
+            {
+                existingQueue.SetPermissions(LocalEveryoneGroupName, MessageQueueAccessRights.GenericWrite, AccessControlEntryType.Revoke);
+                existingQueue.SetPermissions(LocalAnonymousLogonName, MessageQueueAccessRights.WriteMessage, AccessControlEntryType.Revoke);
+            }
+
+            var creator = new QueueCreator(true);
+            var bindings = new QueueBindings();
+
+            bindings.BindReceiving(testQueueName);
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+
+            var queue = GetQueue(testQueueName);
+
+            MessageQueueAccessRights? everyoneAccessRights;
```

Comment:
@andreasohlund Rename `everyoneAccessRights` to `accessRights` or `nullBecauseRevoked`?